### PR TITLE
Add support for VK_KHR_device_group 

### PR DESCRIPTION
**Pull request by AWoloszyn**

### CODE REVIEW

File: gapii/cc/vulkan_extras.cpp [hunk 1]
```diff
@@ -749,6 +749,46 @@ uint32_t VulkanSpy::SpyOverride_vkEnumerateInstanceExtensionProperties(
...
+        mImports.mVkInstanceFunctions.find(instance);
+  gapii::VulkanImports::PFNVKENUMERATEPHYSICALDEVICEGROUPSKHR next = 
+    inst_func_iter->second.vkEnumeratePhysicalDeviceGroupsKHR;
+
+  auto ret = next(instance, pPhysicalDeviceGroupCount, pPhysicalDeviceGroupProperties);
+  if (!pPhysicalDeviceGroupProperties) {
+    return ret;
+  }
+  for (size_t i = 0; i < *pPhysicalDeviceGroupCount; ++i) {
+    pPhysicalDeviceGroupProperties[i].mphysicalDeviceCount = 1;
```
Comment (RenfengLiu) [inline gapii/cc/vulkan_extras.cpp]: Asking a learning Vulkan question here, why we're setting  `mphysicalDeviceCount` to 1 here?

File: gapis/api/vulkan/api/queued_command_tracking.api [hunk 2]
```diff
@@ -82,8 +82,11 @@ sub void execPendingCommands(VkQueue queue, bool isRoot) {
           fenceObj.Signaled = true
           recordFenceSignal(cmd.SignalFence)
         }
-      }
+      }
```
Comment (RenfengLiu) [inline gapis/api/vulkan/api/queued_command_tracking.api]: Is this an additional with space?

File: gapis/api/vulkan/api/buffer.api [hunk 3]
```diff
@@ -299,11 +311,21 @@ sub void BindBufferMemory2(
...
+                dg.Binding.Bindings[j] = indices[j]
+              }
+            }
+            default: {}
+          }
           next.Ptr = as!VulkanStructHeader*(next.Ptr)[0:1][0].PNext
         }
       }
-      BindBufferMemory(info.buffer, info.memory, info.memoryOffset)
+      BindBufferMemory(info.buffer, info.memory, info.memoryOffset, dg.Binding)
```
Comment (RenfengLiu) [inline gapis/api/vulkan/api/buffer.api]: another learning question, why we are calling `BindBufferMemory` inside `BindBufferMemory2`?

File: gapis/api/vulkan/api/queued_command_tracking.api (hunk 2 above)
Comment (AWoloszyn) [reply gapis/api/vulkan/api/queued_command_tracking.api]: Oops, will fix.

File: gapis/api/vulkan/api/buffer.api (hunk 3 above)
Comment (AWoloszyn) [reply gapis/api/vulkan/api/buffer.api]: So we are calling BindBufferMemory from both `vkBindBufferMemory && vkBindBufferMemory2` a common function so we dont have to duplicate the logic.

File: gapii/cc/vulkan_extras.cpp (hunk 1 above)
Comment (AWoloszyn) [reply gapii/cc/vulkan_extras.cpp]: This i so that we can avoid ACTUAL multi-gpu.

File: gapis/api/vulkan/api/device.api [hunk 4]
```diff
@@ -165,6 +169,16 @@ sub ref!DeviceObject createDeviceObject(const VkDeviceCreateInfo* data) {
           ext := as!VkPhysicalDeviceScalarBlockLayoutFeaturesEXT*(next.Ptr)[0]
           object.PhysicalDeviceScalarBlockLayoutFeaturesEXT = new!PhysicalDeviceScalarBlockLayoutFeaturesEXT(
             ScalarBlockLayout: ext.scalarBlockLayout)
+        case VK_STRUCTURE_TYPE_DEVICE_GROUP_DEVICE_CREATE_INFO: {
```
Comment (hevrard) [inline gapis/api/vulkan/api/device.api]: Missing a `}` to close the previous `case`

**Pull request merged**